tests/debugpy/test_evaluate.py

```diff
@@ -155,21 +160,45 @@ def code_to_debug():
             "__a_test",
             "__b_test",
             "__c_test",
-            "__a_test__",
-            "__b_test__",
-            "__c_test__",
         ]
 
+        special_vars_entry = [v for v in vars if v["name"] == "special variables"][0]
+        special_vars_variables = session.request(
+            "variables",
+            {"variablesReference": special_vars_entry["variablesReference"]},
+        )["variables"]
+        special_vars_variables = [
+            v["name"] for v in special_vars_variables if "_test" in v["name"]
+        ]
+        assert special_vars_variables == ["__a_test__", "__b_test__", "__c_test__"]
+
         # String dict keys must be sorted as strings.
         b_test, = (v for v in vars if v["name"] == "b_test")
         b_test_vars = session.request(
             "variables", {"variablesReference": b_test["variablesReference"]}
         )["variables"]
         var_names = [v["name"] for v in b_test_vars]
         if sys.version_info[:2] >= (3, 6):
-            assert var_names == ["'spam'", "'eggs'", "'abcd'", "__len__"]
+            # Note that the special __len__ we manually create is not added to special variables.
```

Comment:
`VariablePresentationHint` has a flag for generated variables like these, "virtual" - it would be nice to use it for synthetic `__len__`. I'm not sure if that makes any difference in UX in VSCode, but I recall that there is a different icon for it in VS.

Also, this may be confusing in a sense that `__len__` is an actual (class) variable that's expected to be a function. If I understand correctly, now we are going to have one `__len__` under "class variables" that's that, and then also `__len__` in normal variables that is the result of calling `len()` on the value. Way back when we had this in ptvsd 3, the way it was distinguished is using `len()` as the variable name. Since no actual attribute should have `()` in the name, it makes it clear that it's synthetic.